src/MainWindow.vala

```diff
@@ -318,18 +318,34 @@ public class Tasks.MainWindow : Hdy.ApplicationWindow {
         var source = list_row.source;
 
         if (Tasks.Application.model.is_remove_task_list_supported (source)) {
-            Tasks.Application.model.remove_task_list.begin (source, (obj, res) => {
-                try {
-                    Tasks.Application.model.remove_task_list.end (res);
-                } catch (Error e) {
-                    critical (e.message);
-                    show_error_dialog (
-                        _("Deleting the task list failed"),
-                        _("The task list registry may be unavailable or unable to be written to."),
-                        e
-                    );
-                }
-            });
+            var message_dialog = new Granite.MessageDialog.with_image_from_icon_name (
+                _("Really Delete List?"),
+                _("The list '%s' will be completely deleted, including all its tasks.").printf (source.display_name),
```

Comment:
Some wording changes to the new dialog. This is just a suggestion, but I think it better follows the HIG. The standard capitalization is sentence case for the primary message, and I think that placing the list name there instead of the smaller text is clearer about the resulting action (although that is clearly up for debate).


```suggestion
                _("Permanently delete list \"%s\"?").printf (source.display_name),
                _("The list and all its tasks will be permanently deleted."),
```

Reply:
On second thoughts, maybe "permanently" is only necessary in the subtitle.